### PR TITLE
Update scripts to run RISC-V emulator using LIT

### PR DESCRIPTION
**Pull request by ArtemSkrebkov**
# Motivation
Many of the examples from `MLIRVector`, `RVVExperiment` can be used as regression tests. It will help to test them regularly and will save the cost of maintenance.

# Opens
* Not ready to be integrated in CI since RISC-V environment is time-consuming to set up. Would need to have https://github.com/buddy-compiler/buddy-mlir/issues/19 resolved and use the same approach for the environment.
* Currently, if there is no RISC-V environment, the corresponding tests will fail. We could make those tests optional and do not run them if the environment is not available.

### CODE REVIEW

File: tests/lit.cfg.py [hunk 1]
```diff
@@ -45,19 +45,28 @@
 # test_source_root: The root path where tests are located.
 config.test_source_root = os.path.dirname(__file__)
 
-# test_exec_root: The root path where tests should be run.
-config.test_exec_root = os.path.join(config.buddy_obj_root, 'tests')
+# tools_dir: tool directories to be used when testing
 config.buddy_tools_dir = os.path.join(config.buddy_obj_root, 'bin')
+config.qemu_tools_dir = "/home/artemy/workspace/repos/buddy-mlir/thirdparty/qemu/build/riscv64-linux-user"
```
Comment (ArtemSkrebkov) [inline tests/lit.cfg.py:50]: Remove absolute path

File: .github/workflows/TestBuild.yml [hunk 2]
```diff
@@ -27,5 +27,7 @@ jobs:
         with:
           path: llvm
           key: build-${{ steps.llvm-build-dir.outputs.commit }}
+      - name: Build RVV enviroment
+        run: bash ./thirdparty/build-rvv-env.sh
```
Comment (ArtemSkrebkov) [inline .github/workflows/TestBuild.yml:31]: Looks like this script does not satisfy resources restrictions. Logs:
https://pipelines.actions.githubusercontent.com/serviceHosts/a2afae62-3cc1-4bba-8025-9c9e34c59e8a/_apis/pipelines/1/runs/286/signedlogcontent/2?urlExpires=2022-08-03T14%3A35%3A18.3447346Z&urlSigningMethod=HMACV1&urlSignature=wrCuK8O%2BbleXbepZnJqXJKvNXiS3HkcYJwARdsL0Dus%3D

**Comment by meshtag** [comment]
What is the status of this PR? @ArtemSkrebkov

**Comment by ArtemSkrebkov** [comment]
Thanks for reaching out.

Sorry, I was busy with other things. If you think that it still can be useful, I can finalize it in spare time.

**Pull request closed without merging**